### PR TITLE
Add waypoint delete cmd

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
To make it more convenient to control the waypoint, use the same mechanism as `apply` instead of searching for the Gateway resource to delete.

### CODE REVIEW

File: istioctl/cmd/waypoint.go [hunk 1]
```diff
@@ -115,6 +119,38 @@ func waypointCmd() *cobra.Command {
...
+		Use:   "delete",
+		Short: "Delete a waypoint configuration",
+		Long:  "Delete a waypoint configuration from the cluster",
+		Example: `  # Delete a waypoint from the current namespace
+  istioctl x waypoint delete
+  
+  # Delete a waypoint from a specific namespace for a specific service account
+  istioctl x waypoint delete --service-account something --namespace default`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			gw := makeGateway(true)
```
Comment (ymesika) [inline istioctl/cmd/waypoint.go:132]: Why do you need this if you set it again few lines later?

Comment (hanxiaop) [reply istioctl/cmd/waypoint.go:132]: It is used to get the gw name with the same logic like `apply/generate` easily. The get function is just to ensure it is there and print the log if it is not found. I think it is redundent, and I'll remove it.

**Review by howardjohn — APPROVED** [review]

**Comment by GregHanson** [comment]
@howardjohn do we have to worry about deleting a waypoint where a user may have used one of the `name` or `service account` override annotations? Or is that a distant edge case since the gateway resource was not created with `istioctl`?

**Comment by howardjohn** [comment]
I think we could say this command assumes the naming conventions of `istioctl x waypoint` to solve that?

**Review by zhlsunshine — APPROVED** [review]
LGTM

**Comment by howardjohn** [comment]
This broke test: `    bookinfo_test.go:279: timeout while waiting after 41 attempts (last error: cannot fetch pod: no pods fetched)
`

The fetch fails if it gets 0 pods, but we want 0 pods

**Comment by hanxiaop** [comment]
> This broke test: ` bookinfo_test.go:279: timeout while waiting after 41 attempts (last error: cannot fetch pod: no pods fetched)`
> 
> The fetch fails if it gets 0 pods, but we want 0 pods

@howardjohn Thanks, I'm going to check it

**Pull request merged**